cmd/eval.go

```diff
@@ -132,6 +138,14 @@ func validateEvalParams(p *evalCommandParams, cmdArgs []string) error {
 		return errors.New("invalid output format for evaluation")
 	}
 
+	//* check if illegal arguments is passed with unknowns flag
+	regexArr := regexp.MustCompile(`^\[.*\]$`)
+	for _, unknwn := range p.unknowns {
+		if regexArr.MatchString(unknwn) {
+			return errors.New(errIllegalUnknownsArg.Error())
+		}
+	}
+
```

Comment:
I wonder if we can attempt to parse it using [`ast.ParseTerm`](https://pkg.go.dev/github.com/open-policy-agent/opa/ast#ParseTerm), like
```go
t, err := ast.ParseTerm(unknwn)
if err != nil { ... }
```
and then check `t.Value`'s type, like
```go
switch t.Value.(type) {
case ast.Ref: // OK
default:
  return errors.New(errIllegalUnknownsArg.Error())
}
```

That way, we should be able to capture _more illegal inputs_ than using the regexp. For example, proficient rego users _could_ attempt to pass a _set_ of things like `{data.unknown, input.foobar}` using `{..}` which isn't captured by the regular expression.

Reply:
sure...will look into it

Reply:
done